zone/attack.cpp

```diff
@@ -1414,8 +1414,8 @@ void Mob::DoAttack(Mob *other, DamageHitInfo &hit, ExtraAttackOptions *opts, boo
 
 	// check to see if we hit..
 	if (!FromRiposte && other->AvoidDamage(this, hit)) {
-		int strike_through = itembonuses.StrikeThrough + spellbonuses.StrikeThrough + aabonuses.StrikeThrough;
-		if (strike_through && zone->random.Roll(strike_through)) {
+		if (int strike_through = itembonuses.StrikeThrough + spellbonuses.StrikeThrough + aabonuses.StrikeThrough;
```

Comment:
Why was this changed to a condition?

Reply:
"if","switch", and range-based for loop initializers should be used to reduce scope of variables, where the variable is never used after the statement to reduce potential errors.